tests-integration/tests/auction_app.rs

```diff
@@ -1004,3 +1061,532 @@ fn test_auction_app_cw20() {
+        auction_component.clone(),
+        cw721_component.clone(),
+        cw20_component.clone(),
+        second_cw20_component.clone(),
+    ];
+
+    let app = MockAppContract::instantiate(
+        andr.get_code_id(&mut router, "app-contract"),
+        owner,
+        &mut router,
+        "Auction App",
+        app_components.clone(),
+        andr.kernel.addr(),
+        Some(owner.to_string()),
+    );
+    let components = app.query_components(&router);
+    assert_eq!(components, app_components);
+    let cw20: MockCW20 = app.query_ado_by_component_name(&router, cw20_component.name);
+
+    // Mint Tokens
+    let cw721: MockCW721 = app.query_ado_by_component_name(&router, cw721_component.name);
+
+    let mint_msg = mock_quick_mint_msg(2, owner.to_string());
+
+    router
+        .execute_contract(
+            owner.clone(),
+            Addr::unchecked(cw721.addr().clone()),
+            &mint_msg,
+            &[],
+        )
+        .unwrap();
+
+    // Send Token to Auction
+    let auction: MockAuction = app.query_ado_by_component_name(&router, auction_component.name);
+
+    router
+        .execute_contract(
+            owner.clone(),
+            Addr::unchecked(auction.addr().clone()),
+            &mock_authorize_token_address(cw721.addr().clone(), None),
+            &[],
+        )
+        .unwrap();
+
+    let start_time = router.block_info().time.nanos() / MILLISECONDS_TO_NANOSECONDS_RATIO + 100;
+    let receive_msg = mock_start_auction(
+        Some(Milliseconds(start_time)),
+        Milliseconds(start_time + 2),
+        cw20.addr().to_string(),
+        true,
+        None,
+        None,
+        None,
+    );
+
+    let send_msg = mock_send_nft(
+        AndrAddr::from_string("./auction".to_string()),
+        "0".to_string(),
+        to_json_binary(&receive_msg).unwrap(),
+    );
+
+    router
+        .execute_contract(
+            owner.clone(),
+            Addr::unchecked(cw721.addr().clone()),
+            &send_msg,
+            &[],
+        )
+        .unwrap();
+
+    router.set_block(BlockInfo {
+        height: router.block_info().height,
+        time: Timestamp::from_nanos(start_time * MILLISECONDS_TO_NANOSECONDS_RATIO),
+        chain_id: router.block_info().chain_id,
+    });
+
+    // Query Auction State
+    let auction_ids_response: AuctionIdsResponse = router
+        .wrap()
+        .query_wasm_smart(
+            auction.addr().clone(),
+            &mock_get_auction_ids("0".to_string(), cw721.addr().to_string()),
+        )
+        .unwrap();
+
+    assert_eq!(auction_ids_response.auction_ids.len(), 1);
+
+    let auction_id = auction_ids_response.auction_ids.first().unwrap();
+    let auction_state: AuctionStateResponse = router
+        .wrap()
+        .query_wasm_smart(auction.addr().clone(), &mock_get_auction_state(*auction_id))
+        .unwrap();
+
+    assert_eq!(auction_state.coin_denom, cw20.addr().to_string());
+
+    // Now whitelist bidder one
```

Comment:
We haven't permissioned bidding have we? Is this necessary?

Reply:
I added that a while ago to test the permissioning system. I blacklisted the bidder, checked for `Unauthorized` error, whitelisted the same bidder, and finally proceeded with the operations.